Modules/Core/SpatialObjects/include/itkArrowSpatialObject.hxx

```diff
@@ -28,182 +27,162 @@ template< unsigned int TDimension >
 ArrowSpatialObject< TDimension >
 ::ArrowSpatialObject()
 {
-  this->SetDimension(TDimension);
-  this->SetTypeName ("ArrowSpatialObject");
-  this->GetProperty()->SetRed(1);
-  this->GetProperty()->SetGreen(0);
-  this->GetProperty()->SetBlue(0);
-  this->GetProperty()->SetAlpha(1);
-
-  m_Direction.Fill(0);
-  m_Direction[0] = 1; // along the x direction by default
-  m_Position.Fill(0);
-  m_Length = 1;
-
-  this->ComputeBoundingBox();
-}
-
-/** Set the length of the arrow */
-template< unsigned int TDimension >
-void
-ArrowSpatialObject< TDimension >
-::SetLength(double length)
-{
-  m_Length = length;
-  double spacing[TDimension];
-  spacing[0] = m_Length;
-
-  for ( unsigned int i = 1; i < TDimension; i++ )
-    {
-    spacing[i] = 1;
-    }
-  this->SetSpacing(spacing);
-  this->Modified();
+  this->SetTypeName("ArrowSpatialObject");
+  this->GetProperty().SetRed(1);
+  this->GetProperty().SetGreen(0);
+  this->GetProperty().SetBlue(0);
+  this->GetProperty().SetAlpha(1);
+
+  m_DirectionInObjectSpace.Fill(0);
+  m_DirectionInObjectSpace[0] = 1; // along the x direction by default
+  m_PositionInObjectSpace.Fill(0);
+  m_LengthInObjectSpace = 1;
+
+  this->Update();
 }
 
 /** Compute the bounding box */
 template< unsigned int TDimension >
 bool
 ArrowSpatialObject< TDimension >
-::ComputeLocalBoundingBox() const
+::ComputeMyBoundingBox() const
 {
   itkDebugMacro("Computing Rectangle bounding box");
 
-  if ( this->GetBoundingBoxChildrenName().empty()
-       || strstr( typeid( Self ).name(), this->GetBoundingBoxChildrenName().c_str() ) )
-    {
-    PointType pnt = this->GetPosition();
-    PointType pnt2;
-    for ( unsigned int i = 0; i < TDimension; i++ )
-      {
-      pnt2[i] = pnt[i] + m_Length * m_Direction[i];
-      }
+  PointType pnt = this->GetPositionInObjectSpace();
 
-    pnt = this->GetIndexToWorldTransform()->TransformPoint(pnt);
-    pnt2 = this->GetIndexToWorldTransform()->TransformPoint(pnt2);
+  const_cast< typename Superclass::BoundingBoxType * >(
```

Comment:
Maybe these const_casts can be avoided with either:
a) Create non const version of GetMyBoundingBoxInObjectSpace()
b) Make the bounding box protected, instead of private. So derived class can access to it with `this->m_MyBoundingBoxInObjectSpace.SetXXX(pnt)`
Not sure if there is a performance hit on creating protected members. Should be double checked.

Reply:
Since the API is changed a lot throughout this Pull request, I agree with @phcerdan 
that it is a good time to make this API `const` consistent.

Reply:
I will create a protected member function that is non-const:  GetModifiableMyBoundingBox()